pkg/organizations/client.go

```diff
@@ -60,13 +60,41 @@ func (c *Client) init() {
 	}
 }
 
+type OrganizationDomainState string
+
+const (
+	OrganizationDomainPending        OrganizationDomainState = "pending"
+	OrganizationDomainVerified       OrganizationDomainState = "verified"
+	OrganizationDomainFailed         OrganizationDomainState = "failed"
+	OrganizationDomainLegacyVerified OrganizationDomainState = "legacy_verified"
+)
+
+type OrganizationDomainVerificationStrategy string
+
+const (
+	Dns    OrganizationDomainVerificationStrategy = "dns"
+	Manual OrganizationDomainVerificationStrategy = "manual"
+)
+
 // OrganizationDomain contains data about an Organization's Domains.
 type OrganizationDomain struct {
 	// The Organization Domain's unique identifier.
 	ID string `json:"id"`
 
 	// The domain value
 	Domain string `json:"domain"`
+
+	// The Organization's unique identifier.
+	OrganizationID string `json:"organization_id"`
+
+	// Verification state of the domain.
+	State OrganizationDomainState `json:"state"`
+
+	// Strategy used to verify the domain.
+	VerificationStrategy OrganizationDomainVerificationStrategy `json:"verification_strategy,omitempty"`
+
+	// Token used for DNS verification.
+	VerificationToken string `json:"verification_token,omitempty"`
```

Comment:
Based on your thread, is your ultimate goal to have the domain data on the `organization_domain` events only, but for now they also reside on the `organization` events?

Reply:
oh no, I meant that the `OrganizationID` specifically is mostly useful in the context of organization domain events (since in organization events or the organization API you have the ID already from the parent organization object). it's included by the API in all cases though.

for the other fields, i don't expect `VerificationStrategy` or `VerificationToken` to be widely used, but i'm including them for completeness. `State` on the other hand is generally useful in all cases where this struct would be used